query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -483,7 +488,7 @@ private Filter rewriteLeafFilter(
         // syntax in Pinot
         handleSelection(resultSetGroup, rowBuilderList, selectedAttributes);
       } else if (resultSetTypePredicateProvider.isResultTableResultSetType(resultSet)) {
-        handleTableFormatResultSet(resultSetGroup, rowBuilderList);
+        handleTableFormatResultSet(resultSetGroup, rowBuilderList, selectionMap);
```

Comment:
Is the reverse transformation not requires in `handleAggregationAndGroupBy`? What about `handleSelection`? Do we have a test for all three cases? We need to extend those tests with avg_rage to see, we are not missing.

Reply:
Not required now.

Reply:
Do we need test for these three cases now ?